pkg/engine/engine.go

```diff
@@ -675,7 +674,88 @@ func getAddonFuncMap(addon api.KubernetesAddon) template.FuncMap {
 	}
 }
 
-func getContainerAddonsString(properties *api.Properties, sourcePath string) string {
+func getClusterAutoscalerAddonFuncMap(addon api.KubernetesAddon, cs *api.ContainerService) template.FuncMap {
```

Comment:
seems like at some point (not sure if this PR is an appropriate time or not) we should move these addon functions out of the huge engine.go file

Reply:
good idea